api/types/hardware_test.go

```diff
@@ -61,9 +65,11 @@ func TestSetDefaultValues(t *testing.T) {
 						},
 						Type: "test",
 					},
-					CPU: HardwareRequestRange{
-						Min: defaultMinCPU,
-						Max: 8,
+					CPU: CPU{
+						HardwareRequestRange: HardwareRequestRange{
+							Min: defaultMinCPU,
+							Max: 8,
+						},
```

Comment:
why is the maximum 8, is this vcpu cores/threads?

Reply:
I actually don't know why we are using `8` here -- but this is a test, where I guess that's the test data we opted for. 
These sets of lines are just wrapping the existing `HardwareRequestRange` for cpu field in a dedicated object that also carries a `CPUType` field we can use.